src/frontend/src/optimizer/plan_node/logical_scan.rs

```diff
@@ -528,7 +538,14 @@ impl ToStream for LogicalScan {
             )));
         }
         if self.predicate().always_true() {
-            Ok(StreamTableScan::new(self.core.clone()).into())
+            if self.is_cdc_table() {
+                Ok(
+                    StreamTableScan::new_with_chain_type(self.core.clone(), ChainType::CdcBackfill)
```

Comment:
Recommend to introduce a new physical operator `CdcTableScan`, because scanning from RW table, sys table or CDC table are completely different in terms of implementation.

Here the `if self.predicate().always_true()` actaully implies a huge difference in implementation, which should be reflected by different phyiscal operators, instead of such `if-else` branch. 

I guess this might be introduced by `sys table` before, so it's acceptable for me too keep it as is now and do refactor later.

Reply:
Yes, a new `CdcTableScan` would be more explicit. But maybe we can do this refactor later and reuse the `StreamTableScan` right now, it is not that intrusive to the framework.

Reply:
Strong +1 for this, and may also appicable to `LogicalScan`. There seems too many branches for handling the CDC table. 😕

Reply:
Would you please open an issue for this?

Reply:
Let me do it https://github.com/risingwavelabs/risingwave/issues/13146